### ISSUE TITLE
Quest PDF, Show In Preview Option is not working

### ISSUE THREAD

**Issue by PoojaParmar26** [QuestPDF/QuestPDF#478]
We have to view PDFs before generating the PDF file. We have tried adding the below-mentioned code but it's not working for us in WinForm and our application gets freezer.

```
using QuestPDF.Fluent;
using QuestPDF.Helpers;
using QuestPDF.Infrastructure;

// code in your main method
var document = Document.Create(container =>
{
    container.Page(page =>
    {
        // page content
    });
});

// instead of the standard way of generating a PDF file
document.GeneratePdf("hello.pdf");

// use the following invocation
document.ShowInPreviewer();

// optionally, you can specify an HTTP port to communicate with the previewer host (default is 12500)
document.ShowInPreviewer(12345);
```

Is it possible for QuestPDF to view PDF before generating the file?

**Comment by MarcinZiabek** [comment]
Please check if the `ShowInPreviewer` method invocation throws any exceptions. The application freezes as this is a synchronous operation.

**Comment by PoojaParmar26** [comment]
Yes for that we have added a try catch block but didn't catch any exceptions.

**Comment by jnyrup** [comment]
We had a similar problem in a WPF app, where the previewer wouldn't launch unless it was started a a special thread.
Off the top of my head I can't recall if it had to be on the UI thread or not.

**Comment by benchvondaranch** [comment]
Same Problem here! Any solution will be appreciated.

**Comment by JonPerkinsOldGit** [comment]
Me too. Love this, 24 hours into discovering it and making my first report. Will support when I have more of a handle on it.

I have 2022.12.1 running within a WinUI app. Hangs in Previewer. Before update (from 2022.12.0, previewer didn't load but no hang).

At least I can export to PDF and open that, but I do like the interactive design process while I am still assembling my first report.

**Comment by luis-fss** [comment]
I don't know if it's suitable for you guys, but what works for me is to use this library: 
https://github.com/brminnick/AsyncAwaitBestPractices

The code is as simple as: `document.ShowInPreviewerAsync().SafeFireAndForget();`

**Comment by mikerg87** [comment]
Is there anything required for a console application for the previewer? I tried the above code in a quick program.cs

Z:\learning\src\mike\QuestPDFLearn\Program.cs(18,10): error CS1061: 'Document' does not contain a definition for 'ShowI
nPreviewer' and no accessible extension method 'ShowInPreviewer' accepting a first argument of type 'Document' could be
 found (are you missing a using directive or an assembly reference?) [\QuestPDFLearn.csproj]

if I comment the line : document.ShowInPreviewer();
PDF is generated on file system as expected.

**Comment by bittwiddles** [comment]
@mikerg87 I'm having the same issue. The `ShowInPreviewer()` is not even an available option. 
![image](https://user-images.githubusercontent.com/71159011/226399941-f9ff8790-cea6-44f8-92bb-1475faabb0f9.png)

**Comment by MarcinZiabek** [comment]
> @mikerg87 I'm having the same issue. The `ShowInPreviewer()` is not even an available option. ![image](https://user-images.githubusercontent.com/71159011/226399941-f9ff8790-cea6-44f8-92bb-1475faabb0f9.png)

The QuestPDF Preview is currently available only in dotnet 6 and beyond, as this is the first release with hot-reload capability.

**Comment by dthoeni** [comment]
> report

You need to add "using QuestPDF.Previewer;" to use this extension method

**Comment by MarcinZiabek** [comment]
Good news! The QuestPDF Previewer tools has been replaced with an enhanced QuestPDF Companion App. You can find more details here: [Companion App Usage](https://www.questpdf.com/companion/usage.html). Hope this helps!

Since there has been no recent activity, I’m closing this issue. If you experience a similar problem or have a related request, feel free to open a new GitHub issue. I appreciate your feedback! ❤️
